ibridges/path.py

```diff
@@ -187,7 +192,60 @@ def exists(self) -> bool:
         """Check if the path already exists on the iRODS server."""
         return self.dataobject_exists() or self.collection_exists()
 
-    def walk(self, depth: int):
+    @property
+    def collection(self) -> irods.collection.iRODSCollection:
+        """Instantiate an iRODS collection.
+
+        Parameters
+        ----------
+        session :
+            Session to get the collection from.
+        path : str
+            Name of an iRODS collection.
+
+        Raises
+        ------
+        ValueError:
+            If the path points to a dataobject and not a collection.
+
+        Returns
+        -------
+        iRODSCollection
+            Instance of the collection with `path`.
+
+        """
+        if self.collection_exists():
+            return self.session.irods_session.collections.get(str(self))
+        if self.dataobject_exists():
+            raise ValueError("Error retrieving collection, path is linked to a data object."
+                            " Use get_dataobject instead to retrieve the data object.")
+        raise irods.exception.CollectionDoesNotExist(str(self))
+
+    @property
+    def dataobject(self) -> irods.data_object.iRODSDataObject:
+        """Instantiate an iRODS data object.
+
+        Raises
+        ------
+        ValueError:
+            If the path is pointing to a collection and not a data object.
+
+        Returns
+        -------
+        iRODSDataObject
+            Instance of the data object with `path`.
+
+        """
+        if self.dataobject_exists():
+            return self.session.irods_session.data_objects.get(str(self))
+        if self.collection_exists():
+            raise ValueError("Error retrieving data object, path is linked to a collection."
+                         " Use get_collection instead to retrieve the collection.")
+
+        raise irods.exception.DataObjectDoesNotExist(str(IrodsPath))
+
+
+    def walk(self, depth: Optional[int] = None) -> Iterable[IrodsPath]:
```

Comment:
The depth parameter does not seem to have an effect. But I wonder whether this is still needed. One can filter how deep to iterate when evaluating the results from the generator.

Reply:
Are you sure? I tried it just now, and it seems to work.. Can you give a case where it doesn't work?

Reply:
Ah, now I understand! The depth is relative to the starting collection .... my mistake. You are right it works. Maybe we can make it explicit that the depth is relative to the starting collection ... just in case others have the same thought as me.

Reply:
I have updated the docstring, is it clear now?